extensions/bodyparser-msgpack/src/__tests__/unit/bodyparser.unit.ts

```diff
@@ -0,0 +1,91 @@
+import {Context} from '@loopback/core';
+import {
+  createResolvedRoute,
+  OperationObject,
+  parseOperationArgs,
+  PathParameterValues,
+  Request,
+  RequestBodyObject,
+  RequestBodyParser,
+  Route,
+} from '@loopback/rest';
+import {
+  expect,
+  ShotRequestOptions,
+  stubExpressContext,
+} from '@loopback/testlab';
+import {encode} from 'msgpack-lite';
+import {MsgPackBodyParser} from '../..';
+
+describe('MessagePack body parser', () => {
+  let requestBodyParser: RequestBodyParser;
+  beforeEach(givenRequestBodyParser);
+
+  it('parses body parameter for MessagePack data', async () => {
+    const contentTypes = [
+      'application/msgpack',
+      'application/x-msgpack',
+      'application/subtype+msgpack',
+    ];
+
+    for (const contentType of contentTypes) {
```

Comment:
I like that you are testing different content types :+1:

I am not a fan of the test design you choose:
1. When a test fails, we don't know which content-type was used.
2. When a test fails on the first content type, we will not test other content types, therefore having incomplete picture about what is working and what is not.

A better solution is to use parameterized test cases. Unfortunately, Mocha does not support that out of the box, but the solution is pretty easy.

```ts
  const contentTypes = [
    'application/msgpack',
    'application/x-msgpack',
    'application/subtype+msgpack',
  ];

  for (const contentType of contentTypes) {
    it(`parses "${contentType}" body as MessagePack data`, async () => {
      // the test code
    });
  });
```